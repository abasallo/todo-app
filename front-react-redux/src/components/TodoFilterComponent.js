import React from 'react'
import { Component, PropTypes } from 'react'
import { BottomNavigation, BottomNavigationItem } from 'material-ui'

import Sync from 'react-material-icons/icons/notification/sync'
import CheckBox from 'react-material-icons/icons/toggle/check-box'
import CheckBoxOutlineBlank from 'react-material-icons/icons/toggle/check-box-outline-blank'

import config from '../config'

class TodoFilterComponent extends Component {

    constructor(props) {
        super(props)
        this.state = { selectedItem: 0 }
    }

    select(item) {
        this.setState({ selectedItem: item })
        this.props.onTabSelected(item)
    }

    render() {
        return (
            <BottomNavigation selectedIndex={ this.state.selectedItem }>
                <BottomNavigationItem label={ config.components.filterButtonAllCaption } icon={ <Sync/> } onClick={ () => this.select(0) }/>
                <BottomNavigationItem label={ config.components.filterButtonActiveCaption } icon={ <CheckBoxOutlineBlank/> } onClick={() => this.select(1) }/>
                <BottomNavigationItem label={ config.components.filterButtonCompletedCaption } icon={ <CheckBox/> } onClick={() => this.select(2) }/>
            </BottomNavigation>
        )}
}

TodoFilterComponent.propTypes = { onTabSelected: PropTypes.func.isRequired }

export default TodoFilterComponent
