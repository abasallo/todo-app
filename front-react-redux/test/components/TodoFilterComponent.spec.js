import React from 'react'
import { shallow } from 'enzyme'

import TodoFilterComponent from '../../src/components/TodoFilterComponent'

it('renders properly, default tab selected', () => { expect(shallow(<TodoFilterComponent onTabSelected={ (item) => {} }/>)).toMatchSnapshot() })

it('State should default initially to zero and propagate up accordingly on update', () => {

    const mockedOnTabSelected = jest.fn()
    const mockedSetState = jest.fn()

    const todoFilterComponent = new TodoFilterComponent({ onTabSelected: mockedOnTabSelected })
    todoFilterComponent.setState = mockedSetState

    expect(todoFilterComponent.state.selectedItem).toBe(0)

    todoFilterComponent.select(1)

    expect(mockedOnTabSelected).toBeCalledWith(1)
    expect(mockedSetState).toBeCalledWith({ selectedItem: 1 })
})
