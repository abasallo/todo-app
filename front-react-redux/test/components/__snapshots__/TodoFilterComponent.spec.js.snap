exports[`test renders properly, default tab selected 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <BottomNavigation
    selectedIndex={0}>
    <BottomNavigationItem
        icon={<NotificationSync />}
        label="All"
        onClick={[Function]} />
    <BottomNavigationItem
        icon={<ToggleCheckBoxOutlineBlank />}
        label="Active"
        onClick={[Function]} />
    <BottomNavigationItem
        icon={<ToggleCheckBox />}
        label="Completed"
        onClick={[Function]} />
</BottomNavigation>,
  "nodes": Array [
    <BottomNavigation
      selectedIndex={0}>
      <BottomNavigationItem
            icon={<NotificationSync />}
            label="All"
            onClick={[Function]} />
      <BottomNavigationItem
            icon={<ToggleCheckBoxOutlineBlank />}
            label="Active"
            onClick={[Function]} />
      <BottomNavigationItem
            icon={<ToggleCheckBox />}
            label="Completed"
            onClick={[Function]} />
</BottomNavigation>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TodoFilterComponent
        onTabSelected={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TodoFilterComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onTabSelected": [Function],
        },
        "refs": Object {},
        "state": Object {
          "selectedItem": 0,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <BottomNavigation
          selectedIndex={0}>
          <BottomNavigationItem
                    icon={<NotificationSync />}
                    label="All"
                    onClick={[Function]} />
          <BottomNavigationItem
                    icon={<ToggleCheckBoxOutlineBlank />}
                    label="Active"
                    onClick={[Function]} />
          <BottomNavigationItem
                    icon={<ToggleCheckBox />}
                    label="Completed"
                    onClick={[Function]} />
</BottomNavigation>,
        "_debugID": 2,
        "_renderedOutput": <BottomNavigation
          selectedIndex={0}>
          <BottomNavigationItem
                    icon={<NotificationSync />}
                    label="All"
                    onClick={[Function]} />
          <BottomNavigationItem
                    icon={<ToggleCheckBoxOutlineBlank />}
                    label="Active"
                    onClick={[Function]} />
          <BottomNavigationItem
                    icon={<ToggleCheckBox />}
                    label="Completed"
                    onClick={[Function]} />
</BottomNavigation>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoFilterComponent
    onTabSelected={[Function]} />,
}
`;
